### PR TITLE
create SSP package/struct

### PR DESCRIPTION
**Pull request by afeld**
Builds on https://github.com/opencontrol/fedramp-templater/pull/8. Just some refactoring.

https://github.com/opencontrol/fedramp-templater/compare/hardcoding...ssp-package?w=1

### CODE REVIEW

**Comment by afeld** [comment]
Ready for review!

File: ssp/ssp_suite_test.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"testing"
 )
 
-func TestModels(t *testing.T) {
+func TestSsp(t *testing.T) {
```
Comment (jcscottiii) [inline ssp/ssp_suite_test.go]: `TestSSP`

File: main.go [hunk 2]
```diff
@@ -1,11 +1,11 @@
 package main
 
 import (
-	"fmt"
 	"log"
 	"os"
 	"path/filepath"
 
+	sspPkg "github.com/opencontrol/fedramp-templater/ssp"
```
Comment (jcscottiii) [inline main.go]: Don't alias the package.

File: main.go [hunk 3]
```diff
@@ -20,14 +20,14 @@ func parseArgs() (inputPath, outputPath string) {
 
 func main() {
 	inputPath, outputPath := parseArgs()
-	wordDoc, err := templater.GetWordDoc(inputPath)
+
+	ssp, err := sspPkg.Load(inputPath)
```
Comment (jcscottiii) [inline main.go]: `doc, err := ssp.Load(inputPath)` since the type it returns is of `ssp.Document`

File: ssp/document.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+// UpdateContent modifies the state of the underlying Word document. Note this is purely for bookkeeping in memory, and does not actually make any changes to the file.
+func (s *Document) UpdateContent() {
+	content := s.xmlDoc.String()
+	s.wordDoc.UpdateConent(content)
+}
+
+// CopyTo copies the contents of this Word document to a new file at the provided path.
+func (s *Document) CopyTo(path string) {
+	s.wordDoc.WriteToFile(path, s.Content())
```
Comment (jcscottiii) [inline ssp/document.go]: maybe for another PR, this could fail, would need to return an error

File: templater/templater.go [hunk 5]
```diff
@@ -1,72 +1,25 @@
...
-// findSummaryTables returns the tables for the controls and the control enhancements.
-func findSummaryTables(doc *xml.XmlDocument) (tables []xml.Node, err error) {
-	// find the tables matching the provided headers, ignoring whitespace
-	return doc.Search("//w:tbl[contains(normalize-space(.), 'Control Summary') or contains(normalize-space(.), 'Control Enhancement Summary')]")
-}
-
-func templatizeXMLDoc(doc *xml.XmlDocument) (err error) {
-	tables, err := findSummaryTables(doc)
+// TemplatizeSsp inserts template tags into (i.e. modifies) the provided Ssp.
+func TemplatizeSsp(s *ssp.Document) (err error) {
```
Comment (jcscottiii) [inline templater/templater.go]: TemplatizeSSP

**Comment by afeld** [comment]
Updated!

**Pull request merged**